src/engine/strat_engine/cmd.rs

```diff
@@ -468,16 +468,22 @@ pub fn thin_metadata_size(
         })?
         .read_to_string(&mut output)?;
     if is_ok {
-        Ok(min(
-            THIN_META_MULT_FACTOR
+        let round = block_size - Sectors(1);
```

Comment:
Since the computation, specifically the ```& !*round``` only works when ```block_size``` is a power of 2, could you put a statement about that somewhere? I think an inexpensive assertion would actually be rather hard to get, so just a pre-condition in the comments is probably the best.

Reply:
Resolved.